projects/ui/src/components/Swap/Actions/Swap.tsx

```diff
@@ -667,197 +713,31 @@ const Swap: FC<{}> = () => {
-    if (pathway === Pathway.BEAN_ETH) {
-      console.debug('[handleEstimate] estimating: BEAN <-> ETH');
-      return Farm.estimate(
-        startToken === Eth
-         ? [
-            farm.wrapEth(
-              FarmToMode.INTERNAL
-            ),
-            ...farm.pair.WETH_BEAN(
-              'WETH',
-              FarmFromMode.INTERNAL,
-              _toMode,
-            ),
-         ]
-         : [
-            ...farm.pair.WETH_BEAN(
-              'BEAN',
-              _fromMode,
-              FarmToMode.INTERNAL, // send WETH to INTERNAL
-            ), // amountOut is not exact
-            farm.unwrapEth(
-              FarmFromMode.INTERNAL_TOLERANT  // unwrap WETH from INTERNAL
-            ), // always goes to EXTERNAL because ETH is not ERC20 and therefore not circ. bal. compatible
-         ],
-       [amountIn],
-       forward,
-     );
-    }
-
-    /// BEAN <-> WETH
-    if (pathway === Pathway.BEAN_WETH) {
-      console.debug('[handleEstimate] estimating: BEAN <-> WETH');
-      return Farm.estimate(
-        startToken === Weth
-          ? farm.pair.WETH_BEAN(
-            'WETH',
-            _fromMode,
-            _toMode,
-          )
-          : farm.pair.WETH_BEAN(
-            'BEAN',
-            _fromMode,
-            _toMode,
-          ),
-       [amountIn],
-       forward,
-     );
-    } 
-
-    /// BEAN <-> CRV3 Underlying
-    if (pathway === Pathway.BEAN_CRV3_UNDERLYING) {
-      console.debug('[handleEstimate] estimating: BEAN <-> 3CRV Underlying');
-      return Farm.estimate(
-        [
-          farm.exchangeUnderlying(
-            farm.contracts.curve.pools.beanCrv3.address,
-            _tokenIn.address,
-            _tokenOut.address,
-            _fromMode,
-            _toMode
-          )
-        ],
-        [amountIn],
-        forward,
-      );
-    }
-
-    throw new Error('Unsupported swap mode.');
-  }, [Eth, Weth, farm, getPathway]);
-
   const handleQuote = useCallback<DirectionalQuoteHandler>(
-    (direction) => async (_tokenIn, _amountIn, _tokenOut) => {
-      console.debug('[handleQuote] ', {
+    (direction, swapOperation) => async (__tokenIn, _amountIn, __tokenOut) => {
+      console.debug('[handleQuoteWithSdk] ', {
         direction,
-        _tokenIn,
         _amountIn,
-        _tokenOut
+        swapOperationPath: swapOperation.getDisplay(),
       }); 
       if (!account) throw new Error('Connect a wallet first.');
-      
-      const amountIn = ethers.BigNumber.from(toStringBaseUnitBN(_amountIn, _tokenIn.decimals));
-      const estimate = await handleEstimate(
-        direction === 'forward',
-        amountIn,
-        account,
-        _tokenIn,
-        _tokenOut,
-        FarmFromMode.INTERNAL_EXTERNAL,
-        FarmToMode.EXTERNAL,
-      );
+
+      const forward: Boolean = direction === 'forward';
+
+      const amountIn = forward ? 
+        ethers.BigNumber.from(toStringBaseUnitBN(_amountIn, swapOperation.tokenIn.decimals)) 
```

Comment:
Think these conversions can be simplified. The UI uses a mix of ethers.BigNumber and BigNumber.js along with a bunch of custom conversion and formatting methods, like toStringBaseUnitBN.

All of this was simplified in the SDK. TokenValue is the all encompassing class to use for values.

Would be good to replace all usage of BigNumber with TokenValue as part of this effort.

For ex this code could be:
```
     const amountIn = forward ?
        swapOperation.tokenIn.fromBlockchain( _amountIn.toString())
        : swapOperation.tokenOut.fromBlockchain( _amountIn.toString());
```  
This makes `amountIn` a TokenValue, which is accepted by the `swapOperation.estimate()` method. 

Even further, the `_amountIn` variable here is BigNumber.js. Tracing where that is created and making it a TokenValue there would be ideal.

Reply:
FYI, just chatted with @silochad about this, switching to TokenValue all the way through would involve forking the TokenInput component, so as to not break all other forms that share it. And there's parallel work being done on these so we'll defer till later.

Reply:
Got it, will look into this.

Reply:
OK 👍